osf/utils/migrations.py

```diff
@@ -163,12 +198,334 @@ def ensure_schemas(*args):
+        schema_block_group_key=schema_block_group_key,
+        example_text=example_text
+    )
+
+# Split question multiple choice options into their own blocks
+def split_options_into_blocks(state, rs, question, schema_block_group_key):
+    """
+    For mapping schemas to schema blocks: splits individual multiple choice
+    options into their own schema blocks
+    """
+    for option in question.get('options', []):
+        answer_text = option if isinstance(option, basestring) else option.get('text')
+        help_text = '' if isinstance(option, basestring) else option.get('tooltip', '')
+
+        create_block(
+            state,
+            rs.id,
+            'select-input-option',
+            display_text=answer_text,
+            help_text=help_text,
+            schema_block_group_key=schema_block_group_key,
+        )
+
+def get_registration_response_key(question):
+    """
+    For mapping schemas to schema blocks:
+    Answer ids will map to the user's response
+    """
+    return question.get('qid', '') or question.get('id', '')
+
+def strip_html(string_with_html):
+    """
+    For mapping schemas to schema blocks:
+    Some original schemas have html in the description/help text.  Just replacing this with empty strings.
+    """
+    stripped_html = re.sub('<.*?>', '', string_with_html)
+    return stripped_html
+
+
+def find_title_description_help_example(rs, question):
+    """
+    For mapping schemas to schema blocks:
+    Schemas are inconsistent with regards to the information going into "title",
+    "description", and "help" blocks.
+
+    :returns tuple, title, description, help, example strings
+
+    """
+    title = question.get('title', '')
+    description = strip_html(question.get('description', ''))
+    help = strip_html(question.get('help', ''))
+    example = ''
+
+    schema_name = rs.schema.get('name', '')
+    # Descriptions that contain any of these keywords
+    # are turned into help text instead.
+    help_text_keywords = [
+        'please',
+        'choose',
+        'provide',
+        'format',
+        'describe',
+        'who',
+        'what',
+        'when',
+        'where',
+        'use',
+        'you',
+        'your',
+        'skip',
+        'enter',
+    ]
+
+    if title:
+        if schema_name in ['OSF Preregistration', 'Prereg Challenge']:
+            # These two schemas have clear "example" text in the "help" section
+            example = help
+            help = description
+            description = ''
+        else:
+            for keyword in help_text_keywords:
+                if keyword in description.lower():
+                    help = description
+                    description = ''
+                    break
+    else:
+        # if no title, description text is moved to title.
+        title = description
+        description = ''
+
+    return title, description, help, example
+
+def format_property(question, property, index):
```

Comment:
explain a little more in the docstring -- make it easier to understand at a skim